### ISSUE TITLE
How to train with a custom dataset and apply the Masking operations correctly?

### ISSUE THREAD

**Issue by davimedio01** [khuvll/CAST#5]
How can I use the [“bidir_modeling_crossattn.py”](https://github.com/KHU-VLL/CAST/blob/main/models/bidir_modeling_crossattn.py) code to perform continued training on the ASL Citizen base from the pre-trained model in the K400? I already have Dataset and Dataloaders functions customized for video with PyTorch, but I'd like to know how to apply the masking operations to tubes as well (via the TubeMaskingGenerator class), with the same approch as in your work.

I can already use the code “bidir_modeling_crossattn.py”, believing that this is the one that contains the complete bidirectional model (with CLIP and VideoMAE in the respective Adapters of the architecture), but I realized that the masks are not used for VideoMAE Adapter...

**Comment by jong980812** [comment]
Thank you for your interest in our research. 
Is the ‘ASL Citizen’ you mentioned a dataset? Also, the ‘masking operations’ are used during the VideoMAE pretraining process. If you want to training for down-stream datasets, you don't need to use TubeMaskingGenerator.  Could you provide more details on the issue you’re experiencing? If you’re encountering an error, a screenshot would be helpful for us to better understand the problem.

**Comment by davimedio01** [comment]
I found my problem. In my case, I'm going to do “two” fine-tunings of the model, but since they're for the same task (just by changing the dataset), it turns out that using the model for fine-tuning will be correct, since it's already pre-trained in the k400.

However, I have a question about how to use Deepspeed correctly, since I want to use the class referring to the model in the code of [“bidir_modeling_crossattn.py”](https://github.com/KHU-VLL/CAST/blob/main/models/bidir_modeling_crossattn.py) in my custom code, but I don't know how to start Deepspeed correctly in this case.

**Comment by jong980812** [comment]
So, your dataset issue has been resolved? That’s great to hear!

Regarding your question, when you mention “class,” are you referring to a defined model? If I understand correctly, it seems like you’re facing an issue where the model is not being loaded into DeepSpeed. If possible, please share the error log.

Since I don’t know the exact error details, I can’t provide a precise solution, but one approach is to check whether DeepSpeed is installed correctly. Alternatively, you could try running the model without DeepSpeed to see if that helps.
